### PR TITLE
openssl_csr: allow to specify CRL distribution endpoints

### LINKED ISSUE #147: openssl_csr: Please add ability to specify the crl location (crlDistributionPoints URI http://$url/org.crl)
##### SUMMARY
Please add the ability to specify the crlDistributionPoints URI within the CSR.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
openssl_csr

##### ADDITIONAL INFORMATION
This is needed for smart card logon as a windows domain user (Samba DC).  For this to work, the system looks for the crlDistribution point within the certificate to specify the URL to check for the CRL.

- name: Generate Certificate Authority Certificate Signing Request (CSR)
  openssl_csr:
    path: "/etc/ssl/csr/www.ansible.com.csr"
    privatekey_path: "/etc/ssl/private/ansible.com.pem"
    privatekey_passphrase: "{{ privatekey_passphrase }}"
    common_name: "{{ common_name }}"
    country_name: "{{ country_name }}"
    email_address: "{{ email_address }}"
    organization_name: "{{ organization_name }}"
    key_usage: "{{ item.keyusage }}"
    basic_constraints: "{{ item.basic_constraints }}"
    create_subject_key_identifier: yes
    crl_distribution_point: "http://example.com/pki/ca.crl"
    owner: "{{ ansible_user }}"
    group: "{{ ansible_user }}"

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Resolves #147.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
openssl_csr
openssl_csr_pipe

### CODE REVIEW

**Comment by felixfontein** [comment]
ready_for_review

**Comment by gderber** [comment]
Hi Felix,

Thank you so much for this!  I have applied this pull request as a patch to my 1.3.0 install and it works perfectly.  It's exactly what I needed.

**Comment by felixfontein** [comment]
@gderber thanks, I'm glad it works and solves your problem!

**Comment by felixfontein** [comment]
CC @Andersson007

File: changelogs/fragments/167-openssl_csr-crl-distribution-points.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- "openssl_csr, openssl_csr_pipe - allow to specify CRL distribution endpoints (https://github.com/ansible-collections/community.crypto/issues/147, https://github.com/ansible-collections/community.crypto/pull/167)."
```
Comment (Andersson007) [inline changelogs/fragments/167-openssl_csr-crl-distribution-points.yml:2]:
```suggestion
- "openssl_csr, openssl_csr_pipe - allow to specify CRL distribution endpoints with ``crl_distribution_points`` new option (https://github.com/ansible-collections/community.crypto/issues/147, https://github.com/ansible-collections/community.crypto/pull/167)."
or something like that to mention the option name
```

**Review by Andersson007 — COMMENTED** [review]
@felixfontein thanks for the fix!
could you please add an example to the EXAMPLES block?

Thanks!

File: changelogs/fragments/167-openssl_csr-crl-distribution-points.yml (hunk 1 above)
Comment (felixfontein) [reply changelogs/fragments/167-openssl_csr-crl-distribution-points.yml]: Sorry for editing your suggestion, this way it's still your suggestion and not my own ;)

**Review by Andersson007 — APPROVED** [review]

**Comment by felixfontein** [comment]
@gderber thanks for testing this!
@Andersson007 thanks for reviewing!

**Pull request merged**